database/src/main/java/database/service/SqlDatabaseService.java

```diff
@@ -0,0 +1,438 @@
+            throw new DeletionDatabaseException(UNABLE_DELETE_RECORD);
+        }
+    }
+
+    @Override
+    public void removeAllRecordsFromTable(Class<? extends BaseEntity> entityClass) {
+        String tableName = entityClass.getSimpleName();
+        String deleteAllRecordsSQL = "DELETE FROM " + tableName;
+
+        LOG.info("Executing SQL: {}", deleteAllRecordsSQL);
+
+        try (Connection connection = DriverManager.getConnection(dbURL, dbUserName, dbPassword);
+             Statement statement = connection.createStatement()) {
+            statement.executeUpdate(deleteAllRecordsSQL);
+        } catch (SQLException e) {
+            LOG.error(UNABLE_DELETE_ALL_RECORDS + ": {}", tableName);
+            throw new DeletionDatabaseException(UNABLE_DELETE_ALL_RECORDS);
+        }
+    }
+
+    @Override
+    public <T extends BaseEntity> T getById(Class<? extends BaseEntity> entityClass, Integer id) {
+        String tableName = entityClass.getSimpleName();
+        String selectRecordByIdSQL = "SELECT * FROM " + tableName + " WHERE id = " + id;
+
+        LOG.info("Executing SQL: {}", selectRecordByIdSQL);
+
+        try (Connection connection = DriverManager.getConnection(dbURL, dbUserName, dbPassword);
+             Statement statement = connection.createStatement();
+             ResultSet resultSet = statement.executeQuery(selectRecordByIdSQL)) {
+            if (resultSet.next()) {
+                T entity = (T) entityClass.getDeclaredConstructor().newInstance();
+
+                List<Field> entityFields = getAllFields(entityClass);
+                for (Field field : entityFields) {
+                    field.setAccessible(true);
+                    String columnName = field.getName();
+
+                    Object value = resultSet.getObject(columnName);
+                    field.set(entity, value);
+                }
+                return entity;
+            } else {
+                throw new EntityNotFoundException("Record with id " + id + " not found");
+            }
+        } catch (SQLException | ReflectiveOperationException e) {
+            LOG.error(ENTITY_IS_NOT_FOUND + ": {}", id);
+            throw new DatabaseOperationException("Error retrieving record by id");
+        }
+    }
+
+    @Override
+    public <T extends BaseEntity> Iterable<T> getAllRecordsFromTable(Class<? extends BaseEntity> entityClass) {
+        String tableName = entityClass.getSimpleName();
+        String selectAllRecordsSQL = "SELECT * FROM " + tableName;
+
+        List<T> entities = new ArrayList<>();
+        LOG.info("Executing SQL: {}", selectAllRecordsSQL);
+
+        try (Connection connection = DriverManager.getConnection(dbURL, dbUserName, dbPassword);
+             Statement statement = connection.createStatement();
+             ResultSet resultSet = statement.executeQuery(selectAllRecordsSQL)) {
+            while (resultSet.next()) {
+                T entity = (T) entityClass.getDeclaredConstructor().newInstance();
```

Comment:
Please move single entity creation and filling to a separate method and use it here inside loop and in `getById` method.